x-pack/plugins/security_solution/scripts/run_cypress/parallel_serverless.ts

```diff
@@ -0,0 +1,531 @@
+      : 1;
+    // An index which uniquely identifies this instance of the job
+    const chunkIndex: number = process.env.BUILDKITE_PARALLEL_JOB
+      ? parseInt(process.env.BUILDKITE_PARALLEL_JOB, 10)
+      : 0;
+
+    const nonSkippedSpecsForChunk: string[] = [];
+
+    for (let i = chunkIndex; i < nonSkippedSpecs.length; i += chunksTotal) {
+      nonSkippedSpecsForChunk.push(nonSkippedSpecs[i]);
+    }
+
+    return nonSkippedSpecsForChunk;
+  }
+};
+
+const encode = (str: string): string => Buffer.from(str, 'binary').toString('base64');
+
+const getApiKeyFromElasticCloudJsonFile = () => {
+  const userHomeDir = os.homedir();
+  try {
+    const jsonString = fs.readFileSync(path.join(userHomeDir, '.elastic/cloud.json'), 'utf-8');
+    const jsonData = JSON.parse(jsonString);
+    return jsonData.api_key.qa;
+  } catch (e) {
+    log.info('API KEY could not be found in ');
+    return null;
+  }
+};
+
+// Poller function that is polling every 20s, forever until function is resolved.
+async function poll<T>(
+  fn: () => Promise<T>,
+  retries: number = 200,
+  interval: number = 20000
+): Promise<T> {
+  return Promise.resolve()
+    .then(fn)
+    .catch(async function retry(err: Error): Promise<T> {
+      let remainingRetries = retries;
+      if (remainingRetries-- > 0) {
+        remainingRetries -= 1;
+        return new Promise((resolve) => setTimeout(resolve, interval)).then(fn).catch(retry);
+      }
+      throw err;
+    });
+}
+
+// Method to invoke the create environment API for serverless.
+async function createEnvironment(
+  baseUrl: string,
+  projectName: string,
+  runnerId: string,
+  apiKey: string,
+  ftrConfig: SecuritySolutionDescribeBlockFtrConfig,
+  onEarlyExit: (msg: string) => void
+): Promise<Environment> {
+  log.info(`${runnerId}: Creating environment ${projectName}...`);
+  const environment = {} as Environment;
+  const body = {
+    name: projectName,
+    region_id: DEFAULT_REGION,
+  } as CreateEnvironmentRequestBody;
+
+  const productTypes: ProductType[] = [];
+  ftrConfig?.productTypes?.forEach((t) => {
+    productTypes.push(t as ProductType);
+  });
+  if (productTypes.length > 0) body.product_types = productTypes;
+
+  await axios
+    .post(`${baseUrl}/api/v1/serverless/projects/security`, body, {
+      headers: {
+        Authorization: `ApiKey ${apiKey}`,
+        'Content-Type': 'application/json',
+      },
+    })
+    .then((response) => {
+      environment.name = response.data.name;
+      environment.id = response.data.id;
+      environment.region = response.data.region_id;
+      environment.es_url = `${response.data.endpoints.elasticsearch}:443`;
+      environment.kb_url = `${response.data.endpoints.kibana}:443`;
+      environment.product = response.data.type;
+    })
+    .catch((error) => {
+      onEarlyExit(`${error.code}:${error.data}`);
+    });
+  return environment;
+}
+
+// Method to invoke the delete environment API for serverless.
+async function deleteEnvironment(
```

Comment:
The same comments as for `createEnvironment` are applicable here.

Reply:
Fixed in commit : https://github.com/elastic/kibana/pull/169311/commits/d7613c83c18377ed29ec117643d912a047215e32